services/diddoc/queries/diddoc/utils.go

```diff
@@ -0,0 +1,136 @@
+package diddoc
+
+import (
+	"fmt"
+
+	"github.com/cheqd/did-resolver/types"
+	"github.com/cheqd/did-resolver/utils"
+)
+
+func transformKeyEd25519VerificationKey2018ToEd25519VerificationKey2020(
+	verificationMethod types.VerificationMethod,
+) (types.VerificationMethod, error) {
+	publicKeyMultibase, err := utils.Ed25519VerificationKey2018ToEd25519VerificationKey2020(verificationMethod.PublicKeyBase58)
+	if err != nil {
+		return verificationMethod, err
+	}
+
+	verificationMethod.PublicKeyBase58 = ""
+	verificationMethod.Type = string(types.Ed25519VerificationKey2020)
+	verificationMethod.PublicKeyMultibase = publicKeyMultibase
+
+	return verificationMethod, nil
+}
+
+func transformKeyEd25519VerificationKey2018ToJSONWebKey2020(
+	verificationMethod types.VerificationMethod,
+) (types.VerificationMethod, error) {
+	publicKeyJwk, err := utils.Ed25519VerificationKey2018ToJSONWebKey2020(verificationMethod.PublicKeyBase58)
+	if err != nil {
+		return verificationMethod, err
+	}
+
+	verificationMethod.PublicKeyBase58 = ""
+	verificationMethod.Type = string(types.JsonWebKey2020)
+	verificationMethod.PublicKeyJwk = publicKeyJwk
+
+	return verificationMethod, nil
+}
+
+func transformKeyEd25519VerificationKey2020ToEd25519VerificationKey2018(
+	verificationMethod types.VerificationMethod,
+) (types.VerificationMethod, error) {
+	publicKeyBase58, err := utils.Ed25519VerificationKey2020ToEd25519VerificationKey2018(verificationMethod.PublicKeyMultibase)
```

Comment:
Here the result of function is `interface{}` Let's add type casting here.

Reply:
I think we don't need to cast there, because as a result of the function we are setting the interface type{}.